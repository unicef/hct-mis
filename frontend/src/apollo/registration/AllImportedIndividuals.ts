import { gql } from 'apollo-boost';

export const ALL_IMPORTED_INDIVIDUALS_QUERY = gql`
  query AllImportedIndividuals(
    $after: String
    $before: String
    $first: Int
    $last: Int
    $rdiId: String
  ) {
    allImportedIndividuals(
      after: $after
      before: $before
      first: $first
      last: $last
      rdiId: $rdiId
    ) {
      pageInfo {
        hasNextPage
        hasPreviousPage
        startCursor
        endCursor
      }
      totalCount
      edges {
        cursor
        node {
          ...importedIndividualMinimal
        }
      }
    }
  }
`;
