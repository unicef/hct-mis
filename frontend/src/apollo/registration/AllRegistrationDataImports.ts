import { gql } from 'apollo-boost';

export const ALL_REGISTRATION_DATA_IMPORT_QUERY = gql`
  query AllRegistrationDataImports(
    $after: String
    $before: String
    $first: Int
    $last: Int
    $orderBy: String
    $name_Icontains: String
    $importedBy_Id: UUID
    $status: String
    $importDate: Date
  ) {
    allRegistrationDataImports(
      after: $after
      before: $before
      first: $first
      last: $last
      orderBy: $orderBy
      name_Icontains: $name_Icontains
      importedBy_Id: $importedBy_Id
      status: $status
      importDate: $importDate
    ) {
      pageInfo {
        hasNextPage
        hasPreviousPage
        startCursor
        endCursor
      }
      totalCount
      edges {
        cursor
        node {
            ...registrationFragmentMinimal
        }
      }
    }
  }
`;
