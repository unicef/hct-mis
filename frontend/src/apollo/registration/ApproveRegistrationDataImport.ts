import { gql } from 'apollo-boost';

export const APPROVE_REGISTRATION_DATA_IMPORT_MUTATION = gql`
  mutation ApproveRDI($id: ID!) {
    approveRegistrationDataImport(id: $id) {
      registrationDataImport {
        ...registrationDetailed
      }
    }
  }
`;
