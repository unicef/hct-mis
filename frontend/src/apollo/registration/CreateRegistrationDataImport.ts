import { gql } from 'apollo-boost';

export const CREATE_REGISTRATION_DATA_IMPORT_QUERY = gql`
  mutation registrationXlsxImport(
    $registrationDataImportData: RegistrationXlsxImportMutationInput!
  ) {
    registrationXlsxImport(
      registrationDataImportData: $registrationDataImportData
    ) {
      registrationDataImport {
        id
        name
        dataSource
        datahubId
      }
    }
  }
`;
