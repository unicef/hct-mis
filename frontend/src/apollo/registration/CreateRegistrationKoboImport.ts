import {gql} from 'apollo-boost';

export const CREATE_REGISTRATION_DATA_KOBO_IMPORT_QUERY = gql`
  mutation CreateRegistrationKoboImport(
    $registrationDataImportData: RegistrationKoboImportMutationInput!
  ) {
    registrationKoboImport(
      registrationDataImportData: $registrationDataImportData
    ) {
      registrationDataImport {
        id
        name
        dataSource
        datahubId
      }
      validationErrors
    }
  }
`;
