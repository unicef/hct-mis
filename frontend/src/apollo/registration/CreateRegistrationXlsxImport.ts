import {gql} from 'apollo-boost';

export const CREATE_REGISTRATION_DATA_IMPORT_QUERY = gql`
  mutation CreateRegistrationXlsxImport(
    $registrationDataImportData: RegistrationXlsxImportMutationInput!
  ) {
    registrationXlsxImport(
      registrationDataImportData: $registrationDataImportData
    ) {
      registrationDataImport {
        id
        name
        dataSource
        datahubId
      }
      validationErrors
    }
  }
`;
