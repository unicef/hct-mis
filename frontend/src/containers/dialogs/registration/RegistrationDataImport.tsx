/* eslint-disable react-hooks/exhaustive-deps */
import React, { useCallback, useState } from 'react';
import styled from 'styled-components';
import { useHistory } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  Typography,
  Snackbar,
  SnackbarContent,
  MenuItem,
  Select,
  FormControl,
  InputLabel,
} from '@material-ui/core';
import ExitToAppRoundedIcon from '@material-ui/icons/ExitToAppRounded';
import {
  AllProgramsQuery,
  ProgramNode,
  ProgramStatus,
  useAllProgramsQuery,
  useUpdateProgramMutation,
} from '../../../__generated__/graphql';
import { PROGRAM_QUERY } from '../../../apollo/queries/Program';
import { ALL_PROGRAMS_QUERY } from '../../../apollo/queries/AllPrograms';
import { programCompare } from '../../../utils/utils';
import { useBusinessArea } from '../../../hooks/useBusinessArea';
import { useSnackbarHelper } from '../../../hooks/useBreadcrumbHelper';
import { useDropzone } from 'react-dropzone';
import { Field, Form, Formik } from 'formik';
import { FormikSwitchField } from '../../../shared/Formik/FormikSwitchField';
import { FormikTextField } from '../../../shared/Formik/FormikTextField';
import { FormikSelectField } from '../../../shared/Formik/FormikSelectField';

const DialogTitleWrapper = styled.div`
  border-bottom: 1px solid ${({ theme }) => theme.hctPalette.lighterGray};
`;

const DialogFooter = styled.div`
  padding: 12px 16px;
  margin: 0;
  border-top: 1px solid ${({ theme }) => theme.hctPalette.lighterGray};
  text-align: right;
`;

const DialogDescription = styled.div`
  margin: 20px 0;
  font-size: 14px;
  color: rgba(0, 0, 0, 0.54);
`;
const ComboBox = styled(Select)`
  & {
    min-width: 200px;
  }
`;
const DropzoneContainer = styled.div`
  width: 100%;
  height: 100px;
  background-color: rgba(2, 62, 144, 0.1);
  color: #023e90;
  font-size: 14px;
  font-weight: 500;
  letter-spacing: 0.5px;
  line-height: 16px;
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: ${({ theme }) => theme.spacing(5)}px;
`;

function DropzoneField({
  field,
  form,
  decoratorStart,
  decoratorEnd,
  type,
  precision,
  ...otherProps
}) {
  const onDrop = useCallback((acceptedFiles) => {
    console.log('test');
    field.onChange({
      target: { value: acceptedFiles[0], name: field.name },
    });
  }, []);
  const { getRootProps, getInputProps, acceptedFiles, rootRef } = useDropzone({
    onDrop,
  });
  const acceptedFilename =
    acceptedFiles.length > 0 ? acceptedFiles[0].name : null;
  return (
    <div>
      <DropzoneContainer {...getRootProps()}>
        <input {...getInputProps()} />
        {acceptedFilename || 'UPLOAD FILE'}
      </DropzoneContainer>
    </div>
  );
}

export function RegistrationDataImport(): React.ReactElement {
  const history = useHistory();
  const [open, setOpen] = useState(false);
  const snackBar = useSnackbarHelper();
  const businessArea = useBusinessArea();
  const { data } = useAllProgramsQuery({
    variables: { status: ProgramStatus.Draft },
  });
  const [importType, setImportType] = useState();
  const { t } = useTranslation();

  if (!data || !data.allPrograms) {
    return null;
  }
  const programChoices = data.allPrograms.edges.map((item) => ({
    name: item.node.name,
    value: item.node.id,
  }));
  return (
    <span>
      <Button
        variant='contained'
        color='primary'
        startIcon={<ExitToAppRoundedIcon />}
        onClick={() => setOpen(true)}
      >
        IMPORT
      </Button>
      <Dialog
        open={open}
        onClose={() => setOpen(false)}
        scroll='paper'
        aria-labelledby='form-dialog-title'
      >
        <Formik
          onSubmit={(values) => {
            console.log({ values });
          }}
          initialValues={{}}
        >
          {({ submitForm, values }) => (
            <Form>
              <DialogTitleWrapper>
                <DialogTitle id='scroll-dialog-title'>
                  <Typography variant='h6'>
                    {t('Select File to Import')}
                  </Typography>
                </DialogTitle>
              </DialogTitleWrapper>
              <DialogContent>
                <DialogDescription>
                  Are you sure you want to activate this Programme and push data
                  to CashAssist?
                </DialogDescription>

                <FormControl
                  variant='filled'
                  margin='dense'
                >
                  <InputLabel>Import from</InputLabel>
                  <ComboBox
                    value={importType}
                    variant='filled'
                    label=''
                    onChange={(e) => setImportType(e.target.value)}
                    fullWidth
                  >
                    <MenuItem key='excel' value='excel'>
                      Excel
                    </MenuItem>
                    <MenuItem key='kobo' value='kobo'>
                      Kobo
                    </MenuItem>
                  </ComboBox>
                </FormControl>
                {importType === 'excel' ? (
                  <Field name='file' component={DropzoneField} />
                ) : null}
                <Field
                  name='name'
                  fullWidth
                  label='Name Upload'
                  component={FormikTextField}
                />

                <Field
                  name='program'
                  fullWidth
                  label='Program'
                  choices={programChoices}
                  component={FormikSelectField}
                />
              </DialogContent>
              <DialogFooter>
                <DialogActions>
                  <Button onClick={() => setOpen(false)}>CANCEL</Button>
                  <Button
                    type='submit'
                    color='primary'
                    variant='contained'
                    onClick={() => {
                      submitForm();
                    }}
                  >
                    {t('IMPORT')}
                  </Button>
                </DialogActions>
              </DialogFooter>
            </Form>
          )}
        </Formik>
      </Dialog>
    </span>
  );
}
