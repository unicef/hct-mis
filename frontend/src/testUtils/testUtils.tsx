import React, { ReactElement } from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, RenderOptions } from '@testing-library/react';
import cloneDeep from 'lodash/cloneDeep';
import { TestProviders } from './testProviders';

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
) =>
  render(<BrowserRouter>{ui}</BrowserRouter>, {
    wrapper: TestProviders,
    ...options,
  });

export * from '@testing-library/react';
export { customRender as render };

let m_w = 123456789;
let m_z = 987654321;
const mask = 0xffffffff;

// Takes any integer
export const seed = (i) => {
  m_w = (123456789 + i) & mask;
  m_z = (987654321 - i) & mask;
};

// Returns number between 0 (inclusive) and 1.0 (exclusive),
// just like Math.random().
export const random = () => {
  m_z = (36969 * (m_z & 65535) + (m_z >> 16)) & mask;
  m_w = (18000 * (m_w & 65535) + (m_w >> 16)) & mask;
  let result = ((m_z << 16) + (m_w & 65535)) >>> 0;
  result /= 4294967296;
  return result;
};

export const makeApolloLoadingMock = (mocks) => {
  let mocksWithDelay = cloneDeep(mocks);
  for (const mock of mocksWithDelay) {
    mock.request.delay = 1e21;
  }
  return mocksWithDelay;
};
