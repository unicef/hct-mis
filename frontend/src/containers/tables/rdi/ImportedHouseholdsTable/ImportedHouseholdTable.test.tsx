import { MockedProvider } from '@apollo/react-testing';
import React from 'react';
import { act } from '@testing-library/react';
import wait from 'waait';
import { ImportedHouseholdTable } from '.';
import { render, ApolloLoadingLink } from '../../../../testUtils/testUtils';
import { fakeApolloAllImportedHouseholds } from '../../../../../fixtures/registration/fakeApolloAllImportedHouseholds';

describe('containers/tables/rdi/ImportedHouseholdTable', () => {
  it('should render with data', async () => {
    const { container } = render(
      <MockedProvider mocks={fakeApolloAllImportedHouseholds}>
        <ImportedHouseholdTable
          businessArea='afghanistan'
          rdiId='UmVnaXN0cmF0aW9uRGF0YUltcG9ydE5vZGU6YzY1NzRkODQtMzEzYS00MTNlLTgzMDUtMDY5ZmU4NWMyOGRl'
        />
      </MockedProvider>,
    );
    await act(() => wait(0)); // wait for response

    expect(container).toMatchSnapshot();
  });

  it('should render loading', () => {
    const { container } = render(
      <MockedProvider
        link={new ApolloLoadingLink()}
        mocks={fakeApolloAllImportedHouseholds}
      >
        <ImportedHouseholdTable
          businessArea='afghanistan'
          rdiId='UmVnaXN0cmF0aW9uRGF0YUltcG9ydE5vZGU6YzY1NzRkODQtMzEzYS00MTNlLTgzMDUtMDY5ZmU4NWMyOGRl'
        />
      </MockedProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
