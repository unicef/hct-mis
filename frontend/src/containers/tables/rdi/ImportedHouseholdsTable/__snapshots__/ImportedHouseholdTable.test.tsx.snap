// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/tables/rdi/ImportedHouseholdTable should render loading 1`] = `
<div>
  <div
    class="sc-bdVaJa iMXEqU"
    data-cy="loading-container"
  >
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`containers/tables/rdi/ImportedHouseholdTable should render with data 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="MuiTableContainer-root"
    >
      <div
        class="MuiBox-root MuiBox-root-7"
      >
        <div
          class="MuiBox-root MuiBox-root-8"
        />
      </div>
      <table
        aria-label="enhanced table"
        aria-labelledby="tableTitle"
        class="MuiTable-root makeStyles-table-3"
        data-cy="table-title"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiTableSortLabel-root sc-EHOje dirFCH"
                data-cy="table-label"
                role="button"
                tabindex="0"
              >
                
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiTableSortLabel-root sc-EHOje dirFCH"
                data-cy="table-label"
                role="button"
                tabindex="0"
              >
                Household ID
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiTableSortLabel-root sc-EHOje dirFCH"
                data-cy="table-label"
                role="button"
                tabindex="0"
              >
                Head of Household
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiTableSortLabel-root sc-EHOje dirFCH"
                data-cy="table-label"
                role="button"
                tabindex="0"
              >
                Household Size
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiTableSortLabel-root sc-EHOje dirFCH"
                data-cy="table-label"
                role="button"
                tabindex="0"
              >
                Administrative Level 2
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiTableSortLabel-root sc-EHOje dirFCH"
                data-cy="table-label"
                role="button"
                tabindex="0"
              >
                Data Collected
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root sc-gzVnrw btWNie MuiTableRow-hover"
            data-cy="imported-households-row"
            role="checkbox"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
            />
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
            >
              <div
                class="sc-dnqmqq jbBufH"
              />
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              2
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
            />
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
            >
              <time
                datetime="1610323200000"
              >
                11 Jan 2021
              </time>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="MuiTablePagination-root"
      data-cy="table-pagination"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
      >
        <div
          class="MuiTablePagination-spacer"
        />
        <p
          class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
          id="mui-5806"
        >
          Rows per page:
        </p>
        <div
          class="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
        >
          <div
            aria-haspopup="listbox"
            aria-labelledby="mui-5806 mui-73230"
            class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
            id="mui-73230"
            role="button"
            tabindex="0"
          >
            10
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput"
            tabindex="-1"
            value="10"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
        <p
          class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
        >
          1-10 of 159
        </p>
        <div
          class="MuiTablePagination-actions"
        >
          <button
            aria-label="Previous page"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            title="Previous page"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
            </span>
          </button>
          <button
            aria-label="Next page"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
            tabindex="0"
            title="Next page"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
